amr-wind/projection/incflo_apply_nodal_projection.cpp

```diff
@@ -435,3 +435,119 @@ void incflo::ApplyProjection(
         }
     }
 }
+
+void incflo::UpdateGradP(
```

Comment:
Do you think there is any room to refactor this to extract the common code needed by the nodal projection step here also needed by incflo_apply_nodal_projection? I haven't found much redundancy.

Reply:
I attempted to do this, but it wasn't working. Issues with properly setting up sigma and boundary velocities somehow (compiled fine, some of the reg tests failed, some numerically and others by seg faults). The changes are in my branch "calcGradP_addlcode". Going to merge this as-is.